text/0000-temp-components-lib.md

```diff
@@ -0,0 +1,617 @@
+flexibility in creating "custom designs" that either include/remove specific elements.
+
+**This is the current standard supported and promoted by the React team.**
+
+In this example, the `FieldProvider` would act as the `getFormControlProps`
+Headless-styles helper in managing the state of the field and other children.
+
+This means, any field related component (i.e. Label, Input, ErrorMessage, etc.)
+would naturally inherit the state and all a11y props needed that is managed from
+the Headless-styles helpers (invalid, disabled, readOnly, etc.) and require no
+props of those from the user.
+
+This would also allow us to internally create error logs that would "make it hard
+to break something" which would inform the user when they are missing a FieldProvider.
+
+For a more complex component - see the [Detailed design](#detailed-design) section.
+
+## Motivation
+
+### Why are we doing this?
+
+There has been consistent requests for a React component library due to the nature
+of most developers either being either unaware of how to use a "headless" library
+or generally uncomfortable with being responsible for owning their own components.
+
+Having a component library that has a familiar interface which ultimately uses
+Headless-styles and React-utils under the hood should help break that barrier.
+
+### What use cases does it support?
+
+A component library would be most beneficial to teams/users that:
+
+1. Are mostly Backend developers that have little to no experience using Javascript or React.
+2. Junior frontend developers who have little to no experience in Javascript or React.
+3. Marketing departments who would like to use Pando, but have "just enough experience to be dangerous" in frontend development.
+4. Team leads that haven't been able to keep up with the technology trends but still need to code when possible.
+5. Teams that have the experience and understanding of Javascript and React, but don't have the time that may be required to use an API like Headless-styles (vs. dropping in an abstraction of it - component).
+
+### What is the expected outcome?
+
+The expected outcome of creating a component library should be:
+
+1. More adoption for Pando due to having a library that uses a familar pattern that people are generally more comfortable with.
+2. Should naturally teach users about the semantic token naming convention as it will inheritently use Headless-styles for prop definitions (which uses the same naming convention of the themes for
+consistency).
+3. Provide a developer experience that uses "React best practices" for a result that naturally allows scalability with custom designs via composable components, and high performance via advanced React patterns.
+4. Share the functionality as Lego blocks (create your own outcome based on what you combine/remove).
+5. By nature, it should increase the community contribution for the same reasons
+mentioned above (a familiar design).
+
```

Comment:
We (A-Team) think that a key benefit of this library is UI consistency across the platform and noting this might be important for engineers/product folks to adopt this library. 😊 Perhaps worth calling out?

Reply:
@phoenixfire07 go ahead and add a suggestion with a new 6. point and I'll commit it. 👍